docs/design/architectural-overview.md

```diff
@@ -0,0 +1,148 @@
+# Kuadrant Architectural Overview [Draft]
+
+## Overview
+
+Kuadrant provides a set of policy APIs, that are reconciled by a set of policy controllers and enforced via integration at different points to configure, secure and enhance application connectivity provided via Gateway API. These policy APIs are focused around areas such as DNS, global load balancing and health checks, alongside data plane specific APIs such as rate limiting and auth. Kuadrant builds ontop of the upstream emerging standard for ingress [Gateway API](https://gateway-api.sigs.k8s.io/). From Gateway API it leverages the concept of [policy attachment](https://gateway-api.sigs.k8s.io/references/policy-attachment/) in order to extend the core Gateway API objects with our policy APIs. Kuadrant also integrates with [OCM (Open Cluster Manager)](https://open-cluster-management.io/) as a multi-cluster control plane to enable defining and distributing Gateways across multiple clusters and offers policy APIs that work at this multi-cluster level also. 
+
+
+## Key Architectural Areas
+
+* The Kuadrant architecture is spread across a control plane and also a data plane. Kuadrant can work in both a single and multi-cluster context. Currently for all APIs to work in a single cluster context you need to have OCM installed in that single cluster also.  
+* The control plane is where policies are exposed and expressed as kubernetes APIs and reconciled by the kuadrant policy controllers. 
+* The data plane is where kuadrant provided policy enforcement points are installed and integrated with a running gateway. 
+* Other policies such as rate limit policy and auth policy are reconciled in the control plane and converted into configuration to be consumed by the data plane components.
+
+## 1000m Architecture
+
+![](./images/1000m-overview.png)
+
+**Note:**
+Currently some of our policies are focused only around the multi-cluster context (DNSPolicy and TLSPolicy in particular). It is high on our roadmap priority list to ensure our policies can operate at both the single and multi-cluster level.
+
+
+
+### Control Plane Components and Responsibilities
+A control plane component is something responsible for accepting instruction via a CRD based API and ensuring that configuration is manifested into state that can be acted on.
+
+- **[Kuadrant Operator](https://github.com/Kuadrant/kuadrant-operator):**
+    - Installation of data plane components via their respective operators
+    - Exposes RateLimitPolicy and AuthPolicy and is currently the policy controller for these APIs
+    - Configures the Gateway to be able to leverage the data plane enforcement components
+- **[Multi-Cluster Gateway Controller](https://github.com/Kuadrant/multicluster-gateway-controller):**
```

Comment:
not sure I follow? Multi-Cluster gateway controller is part of the current architecture?

Reply:
It is but there was a talk about unifying the operators in the future so I wanted to ask if this doc is supposed to cover these changes now or if it will be part of the future RFC

Reply:
ah ok. I call that out under Future Direction at the end. I am fleshing that out further. The details of how that will happen will end up in an RFC though